File: WorkIt/app/src/main/java/prot3ct/workit/views/my_tasks/base/MyTasksContract.java
package prot3ct.workit.views.my_tasks.base;

import java.util.List;

import prot3ct.workit.base.BaseView;
import prot3ct.workit.view_models.MyTasksListViewModel;

public interface MyTasksContract {
    interface View extends BaseView<Presenter> {
        void showCreateJobActivity();

        void notifySuccessful(String message);

        void filterTask(String query);

        void notifyError(String errorMessage);

        void setupTasksAdapter(final List<MyTasksListViewModel> tasks);
    }

    interface Presenter {
        void getMyTasks();

        void deleteTask(int taskId);
    }
}

File: WorkIt/app/src/main/java/prot3ct/workit/views/my_tasks/base/MyTasksContract.kt
package prot3ct.workit.views.my_tasks.base

import prot3ct.workit.base.BaseView
import prot3ct.workit.view_models.MyTasksListViewModel

interface MyTasksContract {
    interface View : BaseView<Presenter> {
        fun showCreateJobActivity()
        fun notifySuccessful(message: String)
        fun filterTask(query: String)
        fun notifyError(errorMessage: String)
        fun setupTasksAdapter(tasks: List<MyTasksListViewModel>)
    }

    interface Presenter {
        val myTasks: Unit
        fun deleteTask(taskId: Int)
    }
}